docs/hacking-an-agent.md

```diff
@@ -1,6 +1,200 @@
-!!!	Warning
-	Not recommended.
+## Preliminaries
 
-!!! Note
-	Coming soon.
+These instructions detail the Python code you need for running an AEA outside the `cli` tool.
 
+!!!	Note
+	You have already coded up your agent. See the <a href="../aea/skill-guide/" target=_blank>build your own skill guide</a> for a reminder.
+
+
+## Imports
+
+First, import the necessary common Python libraries and classes.
+
+``` python
+import os
+import tempfile
+import time	
+from pathlib import Path
+from threading import Thread
+import yaml
+```
+
+Then, import the application specific libraries.
+
+``` python
+from aea import AEA_DIR
+from aea.aea import AEA	
+from aea.configurations.base import ProtocolConfig
+from aea.connections.local.connection import LocalNode, OEFLocalConnection
+from aea.crypto.ledger_apis import LedgerApis
+from aea.crypto.wallet import Wallet
+from aea.mail.base import MailBox, Envelope
+from aea.protocols.base import Protocol
+from aea.protocols.default.message import DefaultMessage
+from aea.protocols.default.serialization import DefaultSerializer
+from aea.protocols.fipa.message import FIPAMessage
+from aea.protocols.fipa.serialization import FIPASerializer
+from aea.registries.base import Resources
+from aea.skills.base import Skill
+from .conftest import CUR_PATH, DummyConnection
+```
+
+
+
+## Initialise the agent
+
+Create a node.
+
+``` python
+node = LocalNode()
+```
+
+Initialise a `MailBox` object with a public key and grab the private key and add it to a wallet.
+
+``` python
+public_key_1 = "mailbox1"
+mailbox1 = MailBox(OEFLocalConnection(public_key_1, node))
+private_key_pem_path = os.path.join(CUR_PATH, "data", "priv.pem")
+wallet = Wallet({'default': private_key_pem_path})
```

Comment:
The wallet creation should be done first.

Then they can get the public key like so:
`public_key = wallet.public_keys['default']`

Then they can set up a mailbox like so:
`mailbox = MailBox(stub_connection)`

Reply:
True, I would still swap it around as in all but this connection case the public key is needed and derived from the wallet.